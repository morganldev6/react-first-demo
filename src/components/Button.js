import React from "react";

const Button = ({ caption, buttonCb }) => {
  return (
    <button className="button" type="button" onClick={buttonCb}>
      {caption}
    </button>
  );
};

export default Button;
