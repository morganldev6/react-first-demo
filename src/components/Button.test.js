import React from "react";
import { render, cleanup } from "@testing-library/react";

import Button from "./Button";

afterEach(cleanup);

describe("Button to display movie description", () => {
  test("Renders correctly", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });
});
