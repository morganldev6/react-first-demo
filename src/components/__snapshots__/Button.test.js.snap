// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button to display movie description Renders correctly 1`] = `
<div>
  <button
    class="button"
    type="button"
  />
</div>
`;
